### PR TITLE
Pop up

### PR DESCRIPTION
**Pull request by MitchfFirstGit**
### HTML & CSS practice: Google-style Popup

[Demo](https://mitchffirstgit.github.io/Pop-up/) |
[Code base](https://github.com/MitchfFirstGit/frontend-2019-homeworks/tree/Pop-up/submissions/MitchfFirstGit/Pop-up)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/MitchfFirstGit/Pop-up/gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+gulp.task('sass-compile', function() {
+    return gulp.src('./scss/main.scss')
+        .pipe(sass())
+        .pipe(nano())
+        .pipe(gulp.dest('./css/'))
+})
+
+gulp.task('watch', function() {
+    gulp.watch('./scss/main.scss', gulp.series('sass-compile'))
+})
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/gulpfile.js]: Add an empty line at the end of every file.

File: submissions/MitchfFirstGit/Pop-up/scss/main.scss [hunk 2]
```diff
@@ -0,0 +1,286 @@
+$nav-border-color: #ccc;
+$btn-bg-color:#ebebeb;
+
+@mixin visually-hidden {
+    position: absolute !important;
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/scss/main.scss]: Avoid `!important` rules.

File: submissions/MitchfFirstGit/Pop-up/css/main.css [hunk 3]
```diff
@@ -0,0 +1 @@
...amboard{background-position:-18px -1849px}.nav li .container .items-container .additional-items li .classroom,.nav li .container .items-container .main-items li .classroom{background-position:-10px -670px}.nav li .container .items-container .additional-items li .earth,.nav li .container .items-container .main-items li .earth{background-position:-10px -2402px}.nav li .container .items-container .additional-items li .collections,.nav li .container .items-container .main-items li .collections{background-position:-15px -1535px}.nav li a{text-decoration:none;display:block}.nav li a:focus{outline:2px solid #ccc}.nav .bell{background-color:#000;border-radius:50%;width:30px;height:29px;opacity:.55;display:flex;justify-content:center;align-items:center;position:relative;bottom:5px}.nav .bell:hover{opacity:1}.nav .bell a{background:url(../img/icons_sprite.png) no-repeat -35px -1356px;width:23px;height:26px}.nav .account{background-color:#8e00a2;font-size:1.5rem;border-radius:50%;width:45px;height:45px;display:flex;justify-content:center;align-items:center;position:relative;bottom:12px;opacity:.55}.nav .account:hover{opacity:1}.nav .account a{color:#fff;font-weight:700;font-family:sans-serif}
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/css/main.css]: Please don't commit compiled or minified files.

File: submissions/MitchfFirstGit/Pop-up/scss/main.scss [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+        .show-hide-container-btn {
+            background: url('../img/icons_sprite.png') no-repeat -7px -1808px;
+            height: 16px;
+            width: 16px;
+            display: inline-block;
+            cursor: pointer;
+            opacity: 0.55;
+        }
+
+        .container {
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/scss/main.scss]: Your class nesting creates too big and specific selector. Read about that [Understand Your Selectors](https://thenextweb.com/2011/03/22/writing-efficient-css-understand-your-selectors/)

File: submissions/MitchfFirstGit/Pop-up/scss/main.scss [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+        justify-content: center;
+        align-items: center;
+        position: relative;
+        bottom: 5px;
+
+        &:hover {
+            opacity: 1;
+        }
+
+        a {
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/scss/main.scss]: Avoid styling by html tag.

File: submissions/MitchfFirstGit/Pop-up/gulpfile.js (hunk 1 above)
Comment (MitchfFirstGit) [reply submissions/MitchfFirstGit/Pop-up/gulpfile.js]: I am very thankful for your review. Please check my code one more time when you will have free time. Thank you for your time and have a nice day!

File: submissions/MitchfFirstGit/Pop-up/scss/main.scss (hunk 2 above)
Comment (MitchfFirstGit) [reply submissions/MitchfFirstGit/Pop-up/scss/main.scss]: I am very thankful for your review. Please check my code one more time when you will have free time. Thank you for your time and have a nice day!

File: submissions/MitchfFirstGit/Pop-up/scss/main.scss (hunk 5 above)
Comment (MitchfFirstGit) [reply submissions/MitchfFirstGit/Pop-up/scss/main.scss]: I am very thankful for your review. Please check my code one more time when you will have free time. Thank you for your time and have a nice day!

File: submissions/MitchfFirstGit/Pop-up/scss/main.scss (hunk 4 above)
Comment (MitchfFirstGit) [reply submissions/MitchfFirstGit/Pop-up/scss/main.scss]: This is a very useful article, now I see my problem. Thank you for your time

File: submissions/MitchfFirstGit/Pop-up/css/main.css [hunk 6]
```diff
@@ -1 +1,209 @@
...
\ No newline at end of file
+.nav {
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/css/main.css]: This file is compiled, remove him. Your PR must contain only the source files.

File: submissions/MitchfFirstGit/Pop-up/gulpfile.js [hunk 7]
```diff
@@ -1,14 +1,14 @@
 const gulp = require('gulp');
 const sass = require('gulp-sass');
-const nano = require('gulp-cssnano');
+// const nano = require('gulp-cssnano');
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/gulpfile.js]: Uncomment this task.

File: submissions/MitchfFirstGit/Pop-up/scss/main.scss [hunk 8]
```diff
@@ -20,267 +9,253 @@ $btn-bg-color:#ebebeb;
...
-            color: white;
-            font-weight: bold;
-            font-family: sans-serif;
-        }
+.show-additional-items,
+.show-hide-container,
+.show-additional-items:checked+.show-additional-items-btn {
+    position: absolute;
+    clip: rect(1px 1px 1px 1px);
+    clip: rect(1px, 1px, 1px, 1px);
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/scss/main.scss]: Doubling properties. Also, CSS Tricks recommended `clip: rect(0 0 0 0);`

File: submissions/MitchfFirstGit/Pop-up/scss/main.scss [hunk 9]
```diff
@@ -20,267 +9,253 @@ $btn-bg-color:#ebebeb;
...
+    width: 30px;
+    height: 29px;
+    opacity: 0.55;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    position: relative;
+    bottom: 5px;
+
+    a {
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/scss/main.scss]: Better use `.class` for this link and selector will be less specific.

File: submissions/MitchfFirstGit/Pop-up/scss/main.scss [hunk 10]
```diff
@@ -20,267 +9,253 @@ $btn-bg-color:#ebebeb;
...
+    width: 45px;
+    height: 45px;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    position: relative;
+    bottom: 12px;
+    opacity: 0.55;
+
+    a {
```
Comment (A-Ostrovnyy) [inline submissions/MitchfFirstGit/Pop-up/scss/main.scss]: Better use `.class` for this link and selector will be less specific.

**Comment by A-Ostrovnyy** [comment]
Don't forget to mark resolved conversations.

**Comment by A-Ostrovnyy** [comment]
Nice work!

**Pull request merged**